### PR TITLE
jupyter,ngcore: implement display data in execute_request

### PR DESCRIPTION
**Pull request by sbinet**
This CL implements displaying data (such as images, PDF, ...) as part of
an execute_request, as part of the execute_result payload.

Updates neugram/ng#128.

### CODE REVIEW

**Comment by sbinet** [comment]
this needs neugram/ng#126 to go in first.

File: jupyter/internal/mimesniff/sniff.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
+// Copyright 2017 The Neugram Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// the following has been extracted from the Go stdlib at revision:
```
Comment (crawshaw) [inline jupyter/internal/mimesniff/sniff.go]: Why are you extracting this from the go stdlib? You could use http.DetectContentType directly.

File: ngcore/ngcore.go [hunk 2]
```diff
@@ -104,79 +105,49 @@ func (n *Neugram) GetOrNewSession(ctx context.Context, name string) *Session {
 	return s
 }
 
-func (s *Session) Exec(src []byte) error {
+func (s *Session) Exec(src []byte) ([][]reflect.Value, error) {
```
Comment (crawshaw) [inline ngcore/ngcore.go]: Can this be collapsed into a []reflect.Value and still be displayed correctly? (That would avoid creating a slice in the common case of src being a single statement.)

File: jupyter/internal/mimesniff/sniff.go (hunk 1 above)
Comment (sbinet) [reply jupyter/internal/mimesniff/sniff.go]: I was worried about importing the whole 'http' package for that one function.
I'll just import it.

File: ngcore/ngcore.go (hunk 2 above)
Comment (sbinet) [reply ngcore/ngcore.go]: Ok.
Actually, as the jupyter backend is only interested in printing/evaluating the last statement, we could just return the last slice.

File: jupyter/internal/mimesniff/sniff.go (hunk 1 above)
Comment (sbinet) [reply jupyter/internal/mimesniff/sniff.go]: done.

File: ngcore/ngcore.go (hunk 2 above)
Comment (sbinet) [reply ngcore/ngcore.go]: done.

**Comment by sbinet** [comment]
PTAL

**Comment by sbinet** [comment]
(I'll squash `5df9b8b` and `a288021` together before merging)

File: ngcore/ngcore.go [hunk 3]
```diff
@@ -104,79 +105,51 @@ func (n *Neugram) GetOrNewSession(ctx context.Context, name string) *Session {
...
-				fmt.Fprint(stdout, format.Debug(v))
-			}
-		}
-		if len(v) > 1 {
-			fmt.Fprintln(stdout, ")")
-		} else if len(v) == 1 {
-			fmt.Fprintln(stdout, "")
+			return nil, Error{Phase: "eval", List: []error{err}}
 		}
+		out = v
```
Comment (crawshaw) [inline ngcore/ngcore.go]: Does this need to be an append if len(out) > 0?

Comment (sbinet) [reply ngcore/ngcore.go]: no, we piggy back on what IPython does: we just return the last statement.
(we can revisit afterwards, if needed)

**Pull request merged**